account_cash_discount_base/models/account_invoice.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+##############################################################################
+#
+#    Authors: Adrien Peiffer
+#    Copyright (c) 2014 Acsone SA/NV (http://www.acsone.eu)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+from datetime import datetime, timedelta
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class account_invoice(models.Model):
+    _inherit = 'account.invoice'
+
+    discount_percent = fields.Float(string='Discount Percent',
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    discount_amount = fields.Float(string='Amount Discount deducted',
+                                   readonly=True,
+                                   states={'draft': [('readonly', False)]})
+    discount_delay = fields.Integer(string='Discount Delay (days)',
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    discount_due_date = fields.Date(string='Discount Due Date',
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+
+    @api.v8
```

Comment:
I think it is not really necessary but it shows that this method is designed for use on new API consistent with the description of the decorator. does not hesitate to correct me if I'm wrong.